src/init.cpp

```diff
@@ -904,9 +904,9 @@ bool AppInitParameterInteraction()
         InitWarning(strprintf(_("Reducing -maxconnections from %d to %d, because of system limitations."), nUserMaxConnections, nMaxConnections));
 
     // ********************************************************* Step 3: parameter-to-internal-flags
-    if (mapMultiArgs.count("-debug") > 0) {
+    if (argsGlobal.IsArgSet("-debug")) {
         // Special-case: if -debug=0/-nodebug is set, turn off debugging messages
-        const std::vector<std::string>& categories = mapMultiArgs.at("-debug");
+        const std::vector<std::string>& categories = argsGlobal.ArgsAt("-debug");
```

Comment:
You can. The lifetime of the temporary is extended by taking a reference to it. This code is effectively equivalent to:
```c++
const std::vector<std::string> categories_tmp = argsGlobal.ArgsAt("-debug");
const std::vector<std::string>& categories = categories_tmp;
```